use formatToParts

gombosg opened the pull request:
Trying to fix #2 - what do you think?

Commit:
use formatToParts

gregolai commented on index.js:
```diff
@@ -38,11 +37,10 @@ function getPseudoISO (date, timeZone) {
-  return str.replace(' ', 'T') + 'Z'
+  }, 'en-US', timeZone).formatToParts(date)
+  const find = (key) => formattedParts.find(part => part.type === key).value
+  const utcDate = `${find('year')}-${find('month')}-${find('day')}T${find('hour')}:${find('minute')}:${find('second')}Z`
```
All these `find`s are performance-intensive. You can refactor this to loop through `formattedParts` only once, extracting year/month/day/etc

gombosg replied:
Got it, switched to a `switch`.

gregolai commented on index.js:
```diff
@@ -38,11 +37,10 @@ function getPseudoISO (date, timeZone) {
+  }, 'en-US', timeZone).formatToParts(date)
+  const find = (key) => formattedParts.find(part => part.type === key).value
+  const utcDate = `${find('year')}-${find('month')}-${find('day')}T${find('hour')}:${find('minute')}:${find('second')}Z`
+  return new Date(utcDate)
```
It used to return a string, but now it returns a `Date` instance?

gombosg replied:
True, oops! The error wasn't apparent since it got `new Date`'d in `getOffset`.

gombosg committed:
use formatToParts - review fixes

- improve O(n) performance to O(1) by using switch for formattedParts
- fix getPseudoISO return value

gombosg commented:
Sorry for the late reply, this got lost in my Github feed.
I pushed some fixes, let me know how you like it. I actually just copied these few lines into our production code, but it works great so far.
